src/properties/propertyowner.cpp

```diff
@@ -235,6 +240,13 @@ void PropertyOwner::addProperty(Property* prop) {
         else {
             _properties.push_back(prop);
             prop->setPropertyOwner(this);
+
+            // Validate the uri
+            std::string uriParent = uri();
+            if (!uriParent.empty()) {
```

Comment:
Also, maybe include information about what is being updated. Takes a few seconds to interpret the code and see if we are updating a property or propertyowner. 

Alternatively, what about creating helper functions for updating propertyowner and properties? Thar could enhance readability quite a bit. E.g. here, moving all the code to a function: 
```cpp
void publishEventPropertyUpdated(std::string_view parentUri, std::string_view propertyId) {
    // Validate URI, <TODO: WHY??>
    if (!parentUri.empty()) {
        std::string id = std::format("{}.{}", parentUri, propertyId);
        global::eventEngine->publishEvent<events::EventPropertyTreeUpdated>(id);
    }
}
```

```cpp
void publishEventPropertyOwnerUpdated(std::string_view ownerUri) {
    // Validate URI, <TODO: WHY??>
    if (!ownerUri.empty()) {
        global::eventEngine->publishEvent<events::EventPropertyTreeUpdated>(ownerUri);
    }
}
```